src/java/org/apache/cassandra/db/compaction/CompactionInfo.java

```diff
@@ -132,6 +132,13 @@ public String toString()
         StringBuilder buff = new StringBuilder();
         buff.append(getTaskType());
 
+        if (getTaskId() != null)
+        {
+            buff.append('(')
+                .append(getTaskId())
+                .append(')');
+        }
```

Comment:
@dcapwell @yifan-c I don't think there's any codepath where the task ID can be null. We might as well avoid the check and make this a one-liner. (Even if it is null, it won't make anything explode...)

Reply:
Double checked the call sites. I agree with out that `getTaskId` cannot return null in the current code base. 
Having the null check here mainly to be consistent with the existing output style (wrap with parenthesis only when id is present) and the null check at [CompactionManager.java#L2025-L2026](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/db/compaction/CompactionManager.java#L2025-L2026)

I am OK to just remove the null check. As you mentioned, it does no harm even with its value being null. 

(Maybe the constructor should assert the task ID not to be null. but it is not related to this ticket.)